src/model/installing/PackageLoader.ts

```diff
@@ -38,3 +38,23 @@ export function GetInstallerIdForPlugin(loader: PackageLoader): PackageInstaller
 
     return null;
 }
+
+/**
+ * Return the PackageLoader enum variant from the provided ecosystem-schema package loader name.
+ * @param name The all lowercase, no underscores, schema name of the package loader
+ * @returns The matching PackageLoader variant, if a valid one exists
+ */
+export function GetInstallerVariantFromName(name: string | undefined): PackageLoader | null {
+    // This is an array of possible variants and indicies (0-n) of the PackageLoader type.
+    // We convert the last n/2-n values to lowercase and removing underscores.
+    const keys = Object.keys(PackageLoader);
+    const loaderNames = keys.slice(keys.length / 2);
+    for (const loaderName of loaderNames) {
+        const schemaName = loaderName.toLowerCase().split("_").join("");
+        if (schemaName == name) {
+            return (PackageLoader as any)[loaderName];
+        }
+    }
+
+    return null;
```

Comment:
Seems a bit hacky. I would filter out number-ish keys and use less type casting:


```typescript
    const enumKeys = Object.keys(PackageLoader)
        .filter((key): key is keyof typeof PackageLoader => isNaN(Number(key)));

    for (const loaderName of enumKeys) {
        const schemaName = loaderName.toLowerCase().split("_").join("");
        if (schemaName == name) {
            return PackageLoader[loaderName];
        }
    }

    return null;
```

Reply:
Agreed!